### ISSUE TITLE
1/4 Test #1: libgit2_clar .............................***Exception: SegFault

### ISSUE THREAD

**Issue by ignatenkobrain** [libgit2/libgit2#4299]
### Reproduction steps
Run tests

### Expected behavior
All tests passing

### Actual behavior
```
1: Test command: /builddir/build/BUILD/libgit2-0.26.0/x86_64-redhat-linux-gnu/libgit2_clar "-xonline" "-xclone::local::git_style_unc_paths" "-xclone::local::standard_unc_paths_are_written_git_style"
1: Test timeout computed to be: 9.99988e+06
1: Loaded 338 suites: 
1: Started
1: 
1: apply::fromdiff...................
1: apply::fromfile..........................................
1: attr::file....
1: attr::flags...
1: attr::ignore..............
1: attr::lookup.....
1: attr::repo..........
1: blame::buffer......
1: blame::getters..
1: blame::harder....
1: blame::simple..S......
1: buf::basic...
1: buf::oom..
1: buf::quote...
1: buf::splice.........
1: checkout::binaryunicode..
1: checkout::conflict.................
1: checkout::crlf..............
1: checkout::head.....
1: checkout::icaseSSSSSSSSSSSSS
1: checkout::index.............................
1: checkout::nasty..............................
1: checkout::tree..........................................
1: checkout::typechange..
1: cherrypick::bare...
1: cherrypick::workdir..........
1: clone::empty...
1: clone::nonetwork...................
1: clone::transport.
1: commit::commit.....
1: commit::parent.
1: commit::parse........
1: commit::signature.........
1: commit::write.......
1: config::add..
1: config::backend.
1: config::configlevel....
1: config::global...
1: config::include........
1: config::multivar.........
1: config::new.
1: config::read..................................
1: config::readonly...
1: config::rename...
1: config::snapshot..
1: config::stress......
1: config::validkeyname.
1: config::write.........................
1: core::array.
1: core::bitvec.
1: core::buffer..........................
1: core::copy...
1: core::dirent.........
1: core::encoding..
1: core::env....
1: core::errors........
1: core::features.
1: core::filebuf......S...S.
1: core::ftruncateSS
1: core::futils.S
1: core::hex.
1: core::iconv...
1: core::init...
1: core::link........................
1: core::mkdir......
1: core::oid...
1: core::oidmap..
1: core::opts..
1: core::path................
1: core::pool....
1: core::posix....
1: core::pqueue.....
1: core::rmdir....
1: core::sha1..
1: core::sortedcache...
1: core::stat..
1: core::stream.
1: core::string.....
1: core::strmap....
1: core::strtol..
1: core::structinit.
1: core::useragent.
1: core::vector.........
1: core::zstream....
1: date::date.
1: date::rfc2822....
1: describe::describe..
1: describe::t6120.......
1: diff::binary............
1: diff::blob..................
1: diff::diffiter.........
1: diff::drivers....
1: diff::format::email.........
1: diff::index.......
1: diff::notify.......
1: diff::parse.......
1: diff::patch......
1: diff::pathspec.
1: diff::racediffiter..
1: diff::rename...........................
1: diff::stats...............
1: diff::submodules.........
1: diff::tree...........
1: diff::workdir............................S
1: fetchhead::nonetwork.............
1: filter::blob...
1: filter::crlf......
1: filter::custom......
1: filter::file..
1: filter::ident..
1: filter::query....
1: filter::stream.S
1: filter::wildcard...
1: graph::descendant::of..
1: index::add.
1: index::addall......
1: index::bypath....SSSSS...S
1: index::cache.....
1: index::collision...
1: index::conflicts...........
1: index::crlf.......
1: index::filemodes.....
1: index::inmemory..
1: index::names.......
1: index::nsec...
1: index::racy.........
1: index::read::index....
1: index::read::tree.
1: index::rename..
1: index::reuc..............
1: index::stage..
1: index::tests..........................
1: index::version...
1: iterator::index.......................
1: iterator::tree........................
1: iterator::workdir.................S...........
1: merge::driver............
1: merge::files..........
1: merge::trees::automerge....
1: merge::trees::commits....
1: merge::trees::modeconflict.
1: merge::trees::recursive................
1: merge::trees::renames...
1: merge::trees::treediff.....
1: merge::trees::trivial.............
1: merge::trees::whitespace...
1: merge::workdir::analysis.......
1: merge::workdir::dirty......
1: merge::workdir::recursive..
1: merge::workdir::renames...
1: merge::workdir::setup................................
1: merge::workdir::simple...............
1: merge::workdir::submodules..
1: merge::workdir::trivial.............
1: network::cred...
1: network::fetchlocal............
1: network::matchhost.
1: network::refspecs.....
1: network::remote::createthenload.
1: network::remote::defaultbranch......
1: network::remote::delete...
1: network::remote::isvalidname..
1: network::remote::local...................
1: network::remote::push..
1: network::remote::remotes..........................
1: network::remote::rename.............
1: network::urlparse..................
1: notes::notes..............
1: notes::notesref.
1: object::blob::filter...
1: object::blob::fromstream..
1: object::blob::write...
1: object::cache....
1: object::commit::commitstagedfile..
1: object::lookup......
1: object::lookupbypath....
1: object::message................
1: object::peel......
1: object::raw::chars..
1: object::raw::compare.......
1: object::raw::convert...
1: object::raw::fromstr..
1: object::raw::hash...........
1: object::raw::short...
1: object::raw::size.
1: object::raw::type2string...
1: object::raw::write.......
1: object::shortid.
1: object::tag::list..
1: object::tag::peel...
1: object::tag::read.....
1: object::tag::write..........
1: object::tree::attributes.....
1: object::tree::duplicateentries..
1: object::tree::frompath..
1: object::tree::read..
1: object::tree::update........
1: object::tree::walk...
1: object::tree::write........
1: odb::alternates..
1: odb::backend::multiple.....
1: odb::backend::nobackend.
1: odb::backend::nonrefreshing.........
1: odb::backend::simple.............
1: odb::emptyobjects...
1: odb::foreach....
1: odb::freshen.....
1: odb::loose........
1: odb::mixed...
1: odb::packed...
1: odb::packed::one..
1: odb::sorting..
1: odb::streamwrite...
1: pack::indexer...
1: pack::packbuilder..........
1: pack::sharing.
1: patch::parse....
1: patch::print.........................
1: path::core...............
1: path::win32..........
1: perf::mergeS
1: rebase::abort.......
1: rebase::inmemory....
1: rebase::iterator..
1: rebase::merge...................
1: rebase::setup..............
1: rebase::submodule.
1: refs::branches::create..........
1: refs::branches::delete........
1: refs::branches::ishead......
1: refs::branches::iterator.....
1: refs::branches::lookup....
1: refs::branches::move...........
1: refs::branches::name...
1: refs::branches::remote....
1: refs::branches::upstream..........
1: refs::branches::upstreamname..
1: refs::crashes.
1: refs::create.............
1: refs::createwithlog.
1: refs::delete...
1: refs::dup..
1: refs::foreachglob.....
1: refs::isvalidname..
1: refs::iterator.......
1: refs::list..
1: refs::listall...
1: refs::lookup.....
1: refs::namespaces..
1: refs::normalize......
1: refs::overwrite....
1: refs::pack...
1: refs::peel......
1: refs::races.....
1: refs::read...............
1: refs::reflog::drop.......
1: refs::reflog::reflog..................
1: refs::rename............
1: refs::revparse.................................
1: refs::settargetwithlog.
1: refs::setter....
1: refs::shorthand.
1: refs::transactions......
1: refs::unicode.
1: refs::update.
1: remote::insteadof.....
1: repo::config...
1: repo::discover..........
1: repo::env.
1: repo::getters...
1: repo::hashfile..
1: repo::head......................
1: repo::headtree....
1: repo::init........................S
1: repo::message..
1: repo::new..
1: repo::open.............
1: repo::pathspec..........
1: repo::reservedname.....
1: repo::setters.....
1: repo::shallow....
1: repo::state.............
1: reset::default.....
1: reset::hard......
1: reset::mixed...
1: reset::soft.........
1: revert::bare...
1: revert::workdir...............
1: revwalk::basic...................
1: revwalk::hidecb......
1: revwalk::mergebase..............
1: revwalk::signatureparsing.
1: revwalk::simplify.
1: stash::apply...................
1: stash::drop.......
1: stash::foreach..
1: stash::save................
1: stash::submodules..
1: status::ignore...............................
1: status::renames................
1: status::single..
1: status::submodules...........
1: status::worktree..........................................
1: status::worktree::init........
1: submodule::add....
1: submodule::init...
1: submodule::lookup.........
1: submodule::modify.......
1: submodule::nosubs.....
1: submodule::open...
1: submodule::repository::init.
1: submodule::status.......
1: submodule::update.......
1: threads::basic....
1: threads::diff..
1/4 Test #1: libgit2_clar .............................***Exception: SegFault 30.94 sec
```

### Version of libgit2 (release number or SHA1)
0.26.0

### Operating system(s) tested
Fedora Rawhide

**Comment by ignatenkobrain** [comment]
I can't reliably reproduce this, it just fails from time to time..

**Comment by pks-t** [comment]
This is a known issue, unfortunately. While we have tests for threading, we do not make guarantees that it can be used the way we do in these tests and as such, they are racy. I've tried multiple times to get our code into a state where they are not racy anymore, but didn't get any satisfying results.

Another solution would be to remove these tests, as they contradict our given guarantees regarding threading anyway.

**Comment by tiennou** [comment]
Ref: #4754

**Comment by ethomson** [comment]
Fixed via #4754
